mne/source_estimate.py

```diff
@@ -2639,14 +2639,70 @@ def morph_data_precomputed(subject_from, subject_to, stc_from, vertices_to,
     return stc_to
 
 
+def _spatio_temporal_src_connectivity_vol(src, n_times):
+    assert len(src) == 1  # not a mixed source space
+    from sklearn.feature_extraction import grid_to_graph
+    vertices = np.where(src[0]['inuse'])[0]
+    n_vertices = len(vertices)
+    data = (1 + np.arange(n_vertices))[:, np.newaxis]
+    # XXX these lines can certainly be replaced by direct logic on the source
+    # space, rather than doing a temp STC, volume export, then data grab. Have
+    # you looked? This would also increase the number of volume source spaces
+    # that can be used to those without MRI, like volume source space within
+    # a generic sphere, because as_volume requires MRI vol source space)
+    stc_tmp = VolSourceEstimate(data, vertices, tmin=0., tstep=1.)
+    img = stc_tmp.as_volume(src, mri_resolution=False)
+    img_data = img.get_data()[:, :, :, 0]
+    img_data = img_data.T
```

Comment:
@agramfort why is this `.T` here? It makes it so that the `img_data.shape != src[0]['shape']` (which it otherwise does without the `.T` there)

Reply:
ahh never mind, I see it now (necessary to get `np.array_equal(img_data[mask], data[:, 0])`)